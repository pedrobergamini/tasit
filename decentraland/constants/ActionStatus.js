const DONE = "done";
const MISSING = "missing";
const PENDING = "pending";

export default {
  DONE,
  MISSING,
  PENDING,
};
