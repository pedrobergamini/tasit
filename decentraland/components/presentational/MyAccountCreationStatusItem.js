import React from "react";
import { StyleSheet, Text, View } from "react-native";
import Colors from "@constants/Colors";
import PropTypes from "prop-types";
import { Ionicons } from "@expo/vector-icons";
import {
  responsiveHeight,
  responsiveWidth,
} from "react-native-responsive-dimensions";
import ActionStatus from "@constants/ActionStatus";

export default function MyAccountCreationStatusItem({ name, status }) {
  return (
    <View style={styles.container}>
      {renderIcon(status)}
      <Text style={styles.actionText}>{name}</Text>
    </View>
  );
}

function renderIcon(status) {
  const ICON_SIZE = 20;
  if (status === ActionStatus.DONE) {
    return (
      <Ionicons
        name="md-checkmark"
        size={ICON_SIZE}
        style={styles.actionStatusIcon}
      />
    );
  }
  if (status === ActionStatus.PENDING) {
    return (
      <Ionicons
        name="md-clock"
        size={ICON_SIZE}
        style={styles.actionStatusIcon}
      />
    );
  }
  return (
    <Ionicons
      name="ios-close"
      size={ICON_SIZE}
      style={styles.actionStatusIcon}
    />
  );
}

MyAccountCreationStatusItem.propTypes = {
  name: PropTypes.string.isRequired,
  status: PropTypes.oneOf(Object.values(ActionStatus)).isRequired,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    backgroundColor: Colors.backgroundColor,
    alignItems: "flex-start",
    justifyContent: "flex-start",
    marginTop: responsiveHeight(1),
    height: responsiveHeight(10),
  },
  actionStatusIcon: {
    fontWeight: "800",
  },
  actionText: {
    fontSize: 16,
    fontWeight: "800",
    marginLeft: responsiveWidth(5),
  },
});
