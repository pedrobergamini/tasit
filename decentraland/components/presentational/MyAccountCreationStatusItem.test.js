import React from "react";
import { shallow } from "enzyme";
import ActionStatus from "../../constants/ActionStatus";
import MyAccountCreationStatusItem from "./MyAccountCreationStatusItem";

describe("MyAccountCreationStatusItem", () => {
  it("renders the component with done status", async () => {
    const name = "test";
    expect(
      shallow(
        <MyAccountCreationStatusItem name={name} status={ActionStatus.DONE} />
      )
    ).toMatchSnapshot();
  });

  it("renders the component with missing status", async () => {
    const name = "test";
    expect(
      shallow(
        <MyAccountCreationStatusItem
          name={name}
          status={ActionStatus.MISSING}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders the component with pending status", async () => {
    const name = "test";
    expect(
      shallow(
        <MyAccountCreationStatusItem
          name={name}
          status={ActionStatus.PENDING}
        />
      )
    ).toMatchSnapshot();
  });
});
