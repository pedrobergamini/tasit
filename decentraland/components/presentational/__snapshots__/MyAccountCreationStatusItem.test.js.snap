// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAccountCreationStatusItem renders the component with done status 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "white",
      "flexDirection": "row",
      "height": 133.4,
      "justifyContent": "flex-start",
      "marginTop": 13.34,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="md-checkmark"
    size={20}
    style={
      Object {
        "fontWeight": "800",
      }
    }
  />
  <Component
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "800",
        "marginLeft": 37.5,
      }
    }
  >
    test
  </Component>
</Component>
`;

exports[`MyAccountCreationStatusItem renders the component with missing status 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "white",
      "flexDirection": "row",
      "height": 133.4,
      "justifyContent": "flex-start",
      "marginTop": 13.34,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="ios-close"
    size={20}
    style={
      Object {
        "fontWeight": "800",
      }
    }
  />
  <Component
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "800",
        "marginLeft": 37.5,
      }
    }
  >
    test
  </Component>
</Component>
`;

exports[`MyAccountCreationStatusItem renders the component with pending status 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "white",
      "flexDirection": "row",
      "height": 133.4,
      "justifyContent": "flex-start",
      "marginTop": 13.34,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="md-clock"
    size={20}
    style={
      Object {
        "fontWeight": "800",
      }
    }
  />
  <Component
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "800",
        "marginLeft": 37.5,
      }
    }
  >
    test
  </Component>
</Component>
`;
